### PR TITLE
SNOW-1272458:Behavior for not-found files / empty files should be identical across different file formats

### PR DESCRIPTION
**Pull request by sfc-gh-yuwang**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-1272458

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Comment by sfc-gh-yuwang** [comment]
Hey guys, this PR meant to fix this [JIRA](https://snowflakecomputing.atlassian.net/browse/SNOW-1272458), which includes behavior change. I have heard that there is a process to announce behavior change, can anyone tell me what is the process?

File: src/snowflake/snowpark/dataframe_reader.py [hunk 1]
```diff
@@ -600,11 +600,12 @@ def _infer_schema_for_file_format(
                 drop_tmp_file_format_if_exists_query = (
                     drop_file_format_if_exists_statement(file_format_name)
                 )
-            results = self._session._conn.run_query(infer_schema_query)["data"]
-            if len(results) == 0:
+            stage_files = self._session._conn.run_query(f"LIST '{path}'")["data"]
+            if len(stage_files) == 0:
                 raise FileNotFoundError(
                     f"Given path: '{path}' could not be found or is empty."
                 )
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/dataframe_reader.py:607]: will empty files still flag this error?

Comment (sfc-gh-yuwang) [reply src/snowflake/snowpark/dataframe_reader.py:607]: It should not, this piece of code check whether given file exist and raise exception when it does not exist, empty file will not trigger it

Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/dataframe_reader.py:607]: in that case, we should remove `"or is empty."` from error message.

Comment (sfc-gh-yuwang) [reply src/snowflake/snowpark/dataframe_reader.py]: you are right! removed

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,17 @@
     - udf
 - Added the function `DataFrame.write.csv` to unload data from a ``DataFrame`` into one or more CSV files in a stage.
 
+### Behavior change
+
+- Changed behavior when reading file from stage. When reading a path that does not exist or when there are no files under the path, `FileNotFound` is raised instead of `SnowflakeSQLException`. This behavior change applies to APIs:
+  - snowflake.snowpark.DataFrameReader:
```
Comment (sfc-gh-aalam) [inline CHANGELOG.md:20]: nit: `FileNotFoundError`.

Also you should mention that csv also displays this behavior, we are aligning the other api with csv and fixing the additional error raised by csv for empty files.

**Pull request closed without merging**